Index: goog/graphics/canvasgraphics.js
===================================================================
--- goog/graphics/canvasgraphics.js	(revision 697)
+++ goog/graphics/canvasgraphics.js	(working copy)
@@ -582,10 +582,44 @@
  *
  * @param {string} text The text string to measure.
  * @param {goog.graphics.Font} font The font object describing the font style.
+ *
+ * @return {number} The width in pixels of the text strings.
  */
-goog.graphics.CanvasGraphics.prototype.getTextWidth = goog.abstractMethod;
+goog.graphics.CanvasGraphics.prototype.getTextWidth = function(text, font) {
+  /** @type {goog.dom.DomHelper} */
+  var domHelper = goog.dom.getDomHelper();
+  
+  /**
+   * The ruler is used to measure the pixel width of Strings The style given
+   * allows the ruler to be off the page (out of sight), with the correct font
+   * and style properties.
+   *
+   * @type {Element}
+   */
+  var ruler_ = domHelper.createDom(
+      'div',
+      {
+        style: 'position:absolute; visibility:hidden; font-family:' +
+            font.family + '; font-size:' +
+            font.size + 'px;'
+      });
 
+  // Add the ruler to the dom
+  document.body.appendChild(ruler_);
+  
+  // Set the inner html of the ruler to be the text
+  ruler_.innerHTML = text;
+  
+  /** @type {number} */
+  var result = ruler_.offsetWidth;
+  
+  // Remove the ruler
+  domHelper.removeNode(ruler_);
+  
+  return result;
+};
 
+
 /**
  * Disposes of the component by removing event handlers, detacing DOM nodes from
  * the document body, and removing references to them.
