### Eclipse Workspace Patch 1.0
#P closure-library-readonly
Index: closure/goog/useragent/useragent.js
===================================================================
--- closure/goog/useragent/useragent.js	(revision 329)
+++ closure/goog/useragent/useragent.js	(working copy)
@@ -57,7 +57,13 @@
  */
 goog.userAgent.ASSUME_OPERA = false;
 
+/**
+ * @define {boolean} Whether we know at compile-time that the user agent is
+ * Rhino
+ */
+goog.userAgent.ASSUME_RHINO = false;
 
+
 /**
  * Whether we know the browser engine at compile-time.
  * @type {boolean}
@@ -68,7 +74,8 @@
     goog.userAgent.ASSUME_GECKO ||
     goog.userAgent.ASSUME_MOBILE_WEBKIT ||
     goog.userAgent.ASSUME_WEBKIT ||
-    goog.userAgent.ASSUME_OPERA;
+    goog.userAgent.ASSUME_OPERA ||
+    goog.userAgent.ASSUME_RHINO;
 
 
 /**
@@ -103,6 +110,13 @@
  */
 goog.userAgent.init_ = function() {
   /**
+   * Whether the user agent string denotes a Rhino client
+   * @type {boolean}
+   * @private
+   */
+  goog.userAgent.detectedRhino_ = false;
+  
+  /**
    * Whether the user agent string denotes Opera.
    * @type {boolean}
    * @private
@@ -145,6 +159,7 @@
   if (!goog.userAgent.BROWSER_KNOWN_ &&
       (ua = goog.userAgent.getUserAgentString())) {
     var navigator = goog.userAgent.getNavigator();
+    goog.userAgent.detectedRhino_ = ua.indexOf('Rhino') != -1;
     goog.userAgent.detectedOpera_ = ua.indexOf('Opera') == 0;
     goog.userAgent.detectedIe_ = !goog.userAgent.detectedOpera_ &&
         ua.indexOf('MSIE') != -1;
@@ -152,7 +167,7 @@
         ua.indexOf('WebKit') != -1;
     // WebKit also gives navigator.product string equal to 'Gecko'.
     goog.userAgent.detectedMobile_ = goog.userAgent.detectedWebkit_ &&
-        ua.indexOf('Mobile') != -1;
+        (ua.indexOf('Mobile') != -1 || ua.indexOf('webOs') != -1);
     goog.userAgent.detectedGecko_ = !goog.userAgent.detectedOpera_ &&
         !goog.userAgent.detectedWebkit_ && navigator.product == 'Gecko';
   }
@@ -163,6 +178,12 @@
   goog.userAgent.init_();
 }
 
+/**
+ * Whether the user agent is Rhino
+ * @type {boolean}
+ */
+goog.userAgent.RHINO = goog.userAgent.BROWSER_KNOWN_ ?
+    goog.userAgent.ASSUME_RHINO : goog.userAgent.detectedRhino_;
 
 /**
  * Whether the user agent is Opera.
